filepicker/filepicker.go

```diff
@@ -0,0 +1,364 @@
+package filepicker
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/charmbracelet/bubbles/key"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/dustin/go-humanize"
+)
+
+// New returns a new filepicker model with default styling and key bindings.
+func New() Model {
+	return Model{
+		Path:          ".",
+		Cursor:        ">",
+		selected:      0,
+		ShowHidden:    false,
+		DirAllowed:    false,
+		FileAllowed:   true,
+		AutoHeight:    true,
+		Height:        0,
+		max:           0,
+		min:           0,
+		selectedStack: newStack(),
+		minStack:      newStack(),
+		maxStack:      newStack(),
+		KeyMap:        DefaultKeyMap,
+		Styles:        DefaultStyles,
+	}
+}
+
+// FileSelectedMsg is the msg that is return when a user makes a valid
+// selection on a file.
+type FileSelectedMsg struct {
+	Path string
+}
+
+type errorMsg struct {
+	err error
+}
+
+type readDirMsg []os.DirEntry
+
+const marginBottom = 5
+const fileSizeWidth = 8
+
+// KeyMap defines key bindings for each user action.
+type KeyMap struct {
+	GoToTop  key.Binding
+	GoToLast key.Binding
+	Down     key.Binding
+	Up       key.Binding
+	PageUp   key.Binding
+	PageDown key.Binding
+	Back     key.Binding
+	Enter    key.Binding
+	Select   key.Binding
+}
+
+// DefaultKeyMap defines the default keybindings.
+var DefaultKeyMap = KeyMap{
+	GoToTop:  key.NewBinding(key.WithKeys("g"), key.WithHelp("g", "first")),
+	GoToLast: key.NewBinding(key.WithKeys("G"), key.WithHelp("G", "last")),
+	Down:     key.NewBinding(key.WithKeys("j", "down", "ctrl+n"), key.WithHelp("j", "down")),
+	Up:       key.NewBinding(key.WithKeys("k", "up", "ctrl+p"), key.WithHelp("k", "up")),
+	PageUp:   key.NewBinding(key.WithKeys("K", "pgup"), key.WithHelp("pgup", "page up")),
+	PageDown: key.NewBinding(key.WithKeys("J", "pgdown"), key.WithHelp("pgdown", "page down")),
+	Back:     key.NewBinding(key.WithKeys("h", "backspace", "left", "esc"), key.WithHelp("h", "back")),
+	Enter:    key.NewBinding(key.WithKeys("l", "right", "enter"), key.WithHelp("l", "enter")),
```

Comment:
Here: `Enter` does not mean the `enter` key, rather entering a directory. I think it would be nice to use a better name (something along the lines of explore or something)

Reply:
`enter directory` (or `enter dir` if that's too long)?

Reply:
Maybe `navigate`?

Reply:
How about `open`?

Reply:
maybe `ChangeDirectory` / `cd`?

Reply:
I like `Open` 👍